app/templating/view_context.py

```diff
@@ -88,39 +97,148 @@ def build_view_context_for_question(metadata, schema, answer_store, current_loca
+        'title': get_question_title(rendered_block, answer_store, schema, metadata,
+                                    current_location.group_instance) % dict(total=formatted_total),
+    })
+    return context
 
-    elif block_type == 'SectionSummary':
-        for section in schema.json['sections']:
-            group_ids = (group['id'] for group in section['groups'])
-            if current_location.group_id in group_ids:
-                section_list = [renderer.render(section, **schema_context)]
-                title = section['title']
-                break
 
+def build_view_context_for_summary(schema, section_list, answer_store, metadata,
+                                   csrf_token, block_type, variables):
     summary_rendering_context = build_summary_rendering_context(schema, section_list, answer_store, metadata)
 
     context = {
         'csrf_token': csrf_token,
         'summary': {
             'groups': summary_rendering_context,
             'answers_are_editable': True,
-            'is_view_submission_response_enabled': is_view_submitted_response_enabled(schema.json),
             'summary_type': block_type,
-            'title': title,
         },
         'variables': variables,
     }
     return context
 
 
+def _build_calculated_summary_section_list(schema, rendered_block, group_id):
+    blocks = []
+    for block in schema.blocks:
+        answers_in_block = schema.get_answers_by_id_for_block(block['id'])
+        answers_to_calculate = rendered_block['calculation']['answers_to_calculate']
+
+        answer_matches = set(answers_in_block.keys()) & set(answers_to_calculate)
+        if answer_matches:
+            blocks.append(_remove_unwanted_questions_answers(block, answers_in_block, answer_matches))
+
+    section = {
+        'title': '',
+        'groups': [
+            {
+                'id': group_id,
+                'title': '',
+                'blocks': blocks,
+            },
+        ],
+    }
+
+    return [section]
+
+
+def _remove_unwanted_questions_answers(block, answers_in_block, answer_ids_to_keep):
+    reduced_block = block.copy()
+    questions_to_keep = [
+        answer['parent_id'] for answer in answers_in_block.values() if answer['id'] in answer_ids_to_keep
+    ]
+
+    questions = []
+    for question in reduced_block['questions']:
+        if question['id'] in questions_to_keep:
+            answers_to_keep = [answer for answer in question['answers'] if answer['id'] in answer_ids_to_keep]
+            question['answers'] = answers_to_keep
+            questions.append(question)
+
+    reduced_block['questions'] = questions
+
+    return reduced_block
+
+
+def _get_formatted_total(block, answer_store, group_instance, schema):
+    calculated_total = 0
+    for answer in block['calculation']['answers_to_calculate']:
+        if group_instance > 0 and not schema.answer_is_in_repeating_group(answer):
```

Comment:
Regarding behaviour with repeats:
1) Do we know that this should always be for the current `group_instance`? 
2) This will not return results a value for repeating answers either?

I can't see anything in the ticket/acceptance criteria around this, so thought we should clear up the expected behaviour